src/box/tuple_format.h

```diff
@@ -281,6 +293,8 @@ struct tuple_format {
 	struct tuple_constraint *constraint;
 	/** Number of constraints. */
 	uint32_t constraint_count;
+	/** Number of fields with a default value. */
+	uint32_t field_defaults_count;
```

Comment:
OK, but `last_field_number` isn't very handy, because it has to be signed to store -1, and it will be used as an upper bound for loop iteration, i.e. always incremented by 1. So I stored `last_field_number + 1` in `struct tuple_format` (not sure about a good name for it).